jpos/src/main/java/org/jpos/q2/Q2.java

```diff
@@ -116,7 +116,7 @@
     private String[] args;
     private boolean hasSystemLogger;
     private boolean exit;
-    private Instant startTime;
+    private Duration startTime;
```

Comment:
Pure uptime calculations using `System.nanoTime()` without any epoch based API's are `Duration`s since `System.nanoTime()` is effectively a JVM runtime `Duration` itself. We only need to use `Instant` when we need an epoch.

Reply:
We use `uptime` to get an idea if the system was booted a few hours ago, a few days ago, or a few months ago.

What's the rationale behind having nano precision here?

Reply:
From my understanding `System.nanoTime()` is the correct way to measure uptime since it's monotonic.